### ISSUE TITLE
Doubt on calculation of smooth_l1 loss

### ISSUE THREAD

**Issue by GPhilo** [fizyr/tf-retinanet#46]
In [this line](https://github.com/fizyr/tf-retinanet/blob/dba799b0c2ccfa3c641a7e14e10599d9e4ec3437/tf_retinanet/losses.py#L93), according to the coment the code is supposed to filter out the samples marked as "ignore" state. What it does, instead, is extracting only the positive samples.

Shouldn't the line be: `indices = tf.where(tf.keras.backend.not_equal(anchor_state, -1))` (or an equivalent version using `>= 0`)?

Unless I'm missing something, the network seems to be training only on positive samples for the regression branch.

**Comment by GPhilo** [comment]
Ah, it seems I did miss something. The targets for negative anchors are computed on an arbitrary GT annotation, so it must be ignored (and indeed for bounding box regression makes sense only for positive samples). Sorry, I got confused by the comment.
